cli/command/container/stats.go

```diff
@@ -48,27 +56,28 @@ func NewStatsCommand(dockerCli command.Cli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.BoolVarP(&opts.all, "all", "a", false, "Show all containers (default shows just running)")
-	flags.BoolVar(&opts.noStream, "no-stream", false, "Disable streaming stats and only pull the first result")
-	flags.BoolVar(&opts.noTrunc, "no-trunc", false, "Do not truncate output")
-	flags.StringVar(&opts.format, "format", "", flagsHelper.FormatHelp)
+	flags.BoolVarP(&options.All, "all", "a", false, "Show all containers (default shows just running)")
+	flags.BoolVar(&options.NoStream, "no-stream", false, "Disable streaming stats and only pull the first result")
+	flags.BoolVar(&options.NoTrunc, "no-trunc", false, "Do not truncate output")
+	flags.StringVar(&options.Format, "format", "", flagsHelper.FormatHelp)
+	flags.Var(&options.Filter, "filter", `Filter containers based on conditions provided`)
 	return cmd
 }
 
-// runStats displays a live stream of resource usage statistics for one or more containers.
+// RunStats displays a live stream of resource usage statistics for one or more containers.
 // This shows real-time information on CPU usage, memory usage, and network I/O.
 //
 //nolint:gocyclo
-func runStats(dockerCli command.Cli, opts *statsOptions) error {
-	showAll := len(opts.containers) == 0
+func RunStats(dockerCli command.Cli, options *StatsOptions) error {
+	showAll := len(options.Containers) == 0
 	closeChan := make(chan error)
 
 	ctx := context.Background()
 
 	// monitorContainerEvents watches for container creation and removal (only
 	// used when calling `docker stats` without arguments).
 	monitorContainerEvents := func(started chan<- struct{}, c chan events.Message, stopped <-chan struct{}) {
-		f := filters.NewArgs()
+		f := options.Filter.Value()
```

Comment:
I'm slightly confused here; should the `--filter` option apply to the _events_, or should the filtering be used for _listing_ the containers? (`getContainerList` below)?


For compose, I could imaging we want to get a list of containers that are part of the compose file, or does compose pass a list of container ID's to look at?

Reply:
(I guess I may have to look at all logic more closely)

Reply:
you're right, filter must also apply to `getContainerList`
passing  `--filter label=compose.docker.compose.project=xx` allows to only select containers and events related to a compose project (also could reduce scope to only focus on a service)

Reply:
Was considering if we should make the "getContainerList" function something that can be passed in. Currently it's a bit of an odd combination of either passing in a hard-coded list of containers, or "all" containers, which uses that function.

If we would pass that function in as argument, perhaps that would allow compose (or other consumers) to do whatever logic they want to get the list of containers.

But honestly, haven't given that much thought yet (or looked what that would look like)